packages/fetchai/skills/generic_seller/handlers.py

```diff
@@ -489,21 +501,78 @@ def _handle_unidentified_dialogue(self, oef_search_msg: OefSearchMessage) -> Non
             )
         )
 
+    def _handle_success(
+        self,
+        oef_search_success_msg: OefSearchMessage,
+        oef_search_dialogue: OefSearchDialogue,
+    ) -> None:
+        """
+        Handle an oef search message.
+
+        :param oef_search_success_msg: the oef search message
+        :param oef_search_dialogue: the dialogue
+        :return: None
+        """
+        self.context.logger.info(
+            "received oef_search success message={} in dialogue={}.".format(
+                oef_search_success_msg, oef_search_dialogue
+            )
+        )
+        target_message = cast(
+            OefSearchMessage,
+            oef_search_dialogue.get_message_by_id(oef_search_success_msg.target),
+        )
+        if (
+            target_message.performative
+            == OefSearchMessage.Performative.REGISTER_SERVICE
+        ):
+            if "location" in target_message.service_description.values:
+                registration_behaviour = cast(
+                    GenericServiceRegistrationBehaviour,
+                    self.context.behaviours.service_registration,
+                )
+                registration_behaviour.register_service_personality_classification()
```

Comment:
`register_service_personality_classification` is now called after `register_agent` is successful.